### PR TITLE
[develop] Update documentation with land configurations and general updates

### PR DESCRIPTION
**Pull request by gspetro-NOAA**
## Description
This PR provides the following documentation updates:
- Addition of a Configuration chapter with land and select ATM configuration information (to be expanded in future PRs).
- Update Code/Technical Overview Chapter to include info on levels of support; update directory structure. 
- Update Build/Run chapter to include information on supported HPC-Stack/spack-stack modules.
- Update Build/Run chapter to include information on where to find RT data.
- Update Build/Run chapter w/land config build info and cross-links for all configs.
- Update Build/Run chapter with info on `rt.sh`
- Add terms to glossary.
- Update I/O chapter with info on LND data.
- Remove nems.configure examples from the I/O chapter and into a `samples` directory. In I/O chapter, link to these documents. Users can click on the samples they want to see instead of scrolling through long and potentially irrelevant code samples. 

### Input data additions/changes
- [X] No changes are expected to input data.

### Anticipated changes to regression tests:
- [X] No changes are expected to any regression test. 

## Subcomponents involved:
- [ ] AQM
- [ ] CDEPS
- [ ] CICE
- [ ] CMEPS
- [ ] CMakeModules
- [ ] FV3
- [ ] GOCART
- [ ] HYCOM
- [ ] MOM6
- [ ] NOAHMP
- [ ] WW3
- [ ] stochastic_physics
- [X] none

### Combined with PR's (If Applicable):
N/A

## Commit Queue Checklist:

- [ ] Link PR's from all sub-components involved
- [ ] Confirm reviews completed in sub-component PR's
- [ ] Add all appropriate labels to this PR.
- [ ] Run full RT suite on either Hera/Cheyenne with both Intel/GNU compilers
- [ ] Add list of any failed regression tests to "Anticipated changes to regression tests" section.

## Linked PR's and Issues:
Closes Issue #1763 .

## Testing Day Checklist:

- [ ] This PR is up-to-date with the top of all sub-component repositories except for those sub-components which are the subject of this PR.
- [ ] Move new/updated input data on RDHPCS Hera and propagate input data changes to all supported systems.

### Testing Log (for CM's):
- RDHPCS
  - Intel
    - [ ] Hera
    - [ ] Orion
    - [ ] Jet
    - [ ] Gaea
    - [ ] Cheyenne
  - GNU
    - [ ] Hera
    - [ ] Cheyenne
- WCOSS2
  - [ ] Dogwood/Cactus
  - [ ] Acorn
- CI
  - [ ] Completed
- opnReqTest
  - [ ] N/A
  - [ ] Log attached to comment

### CODE REVIEW

File: doc/UsersGuide/source/CodeOverview.rst
Comment (natalie-perlin) [inline doc/UsersGuide/source/CodeOverview.rst]: @mark-a-potts @gspetro-NOAA _
AWS Docker container uses Intel compilers, I think.

File: doc/UsersGuide/source/CodeOverview.rst
Comment (DavidHuber-NOAA) [inline doc/UsersGuide/source/CodeOverview.rst:36]: Could you add S4 (Intel) as a supported level 2 platform?

File: doc/UsersGuide/source/BuildingAndRunning.rst
Comment (DavidHuber-NOAA) [inline doc/UsersGuide/source/BuildingAndRunning.rst:108]: S4's data is located in `/data/prod/emc.nemspara/RT`.

**Review by DavidHuber-NOAA — CHANGES_REQUESTED** [review]
Just a request to add S4 information.

**Comment by DeniseWorthen** [comment]
@gspetro-NOAA Is there a built version of this documentation you can point us to?

**Comment by gspetro-NOAA** [comment]
@DeniseWorthen Yes! https://gsp-wm.readthedocs.io/en/text-land-8.2/

File: doc/UsersGuide/source/CodeOverview.rst
Comment (gspetro-NOAA) [reply doc/UsersGuide/source/CodeOverview.rst:36]: I think that is up to @jkbk2004 and the rest of the WM Team. I have no objection, but it's not my call. ;)

File: doc/UsersGuide/source/BuildingAndRunning.rst
Comment (gspetro-NOAA) [reply doc/UsersGuide/source/BuildingAndRunning.rst:108]: Here, I was just listing Level 1 platforms. @jkbk2004 Should I add a section for Level 2 platforms? If so, what is the most updated list? I can also do this in the RRFS configs PR this week if we don't get a decision in time for this PR.

File: doc/UsersGuide/source/CodeOverview.rst
Comment (DeniseWorthen) [inline doc/UsersGuide/source/CodeOverview.rst:23]: There is building the model and there is running the model. I think "supported" means not just building? The L1 systems also contain the required fix/input files. So a L2-4 user would need not to figure out not just how to build it on their platform but also obtain all the required supporting files etc.

File: doc/UsersGuide/source/CodeOverview.rst
Comment (DeniseWorthen) [inline doc/UsersGuide/source/CodeOverview.rst]: We probably need a modification here also. There are now two WW3 caps, but neither resides in a directory called esmf. They reside in a (src) subdirectory. Maybe
 
└── (src) -------- NUOPC WW3 caps

File: doc/UsersGuide/source/CodeOverview.rst
Comment (gspetro-NOAA) [reply doc/UsersGuide/source/CodeOverview.rst:23]: @DeniseWorthen I clarified the wording here so that I explicitly say it runs for Levels 1-3. Level 4 platforms are "supported" but not guaranteed to run (they are "build-only"). We know the WM will build there, and we can help users build it, but we don't guarantee it will run. There has been some discussion of eliminating this category (since what's the point if you can't run it?), but we haven't done that officially yet. 
Normally, we don't mention data in relation to the levels of support because users could be running pretty much any case, not just the sample cases/RT cases we provide. So data for the case they're running may not be available in a central location (or at all) even on a Level 1 system. "Support" has more to do with the availability of the stack and the amount of testing done, according to the definitions passed down to me. However, I can see adding a mention of data for the WM definition, since we're primarily supporting regression testing in standalone mode. I've made that change locally and can push it if you feel strongly that we should include it. 
FYI, there are a few places where levels of support are defined for UFS and/or UFS apps (e.g., [here](https://github.com/ufs-community/ufs/wiki/Supported-Platforms-and-Compilers). "Support" unfortunately hasn't been defined anywhere, but as far as I can tell, it basically means that EPIC will work with you if you fit into one of these levels and are having issues. ;)

File: doc/UsersGuide/source/CodeOverview.rst
Comment (gspetro-NOAA) [reply doc/UsersGuide/source/CodeOverview.rst]: @DeniseWorthen Updated! There are a LOT of files in the (src) directory. Do you think there are other ones we should mention besides the caps?

File: doc/UsersGuide/source/BuildingAndRunning.rst
Comment (jkbk2004) [reply doc/UsersGuide/source/BuildingAndRunning.rst:108]: @DavidHuber-NOAA Can we confirm s4 baseline and input files are maintained at /data/prod/emc.nemspara/RT ? If regularly maintained there, I agree with adding S4 to the list level 2 platform?

File: doc/UsersGuide/source/CodeOverview.rst
Comment (DeniseWorthen) [reply doc/UsersGuide/source/CodeOverview.rst]: Looking at the other components, we're generally just pointing to the directory containing the cap. It is just in the WW3 case, the cap resides in the same directory as all the other code. So I think it is OK to just say "src".

File: doc/UsersGuide/source/BuildingAndRunning.rst
Comment (DavidHuber-NOAA) [reply doc/UsersGuide/source/BuildingAndRunning.rst:108]: @jkbk2004 Yes, the baseline and input files are maintained there.  An rsync is performed nightly from Hera to keep them up to date.  Regression tests are also performed at least once monthly.

File: doc/UsersGuide/source/CodeOverview.rst
Comment (DeniseWorthen) [reply doc/UsersGuide/source/CodeOverview.rst:23]: @gspetro-NOAA I think I don't understand exactly how you're using the term 'support', especially the comment you made that you're _primarily supporting regression testing in standalone mode_. Isn't EPIC's job to support RT testing of the fully UWM repo, including all the non-standalone modes? 

In any case, my original comment was more about implying that on L2-L4, once you build the model OK (however a user manages to do that), it would "run". But it can't run w/o available input files. So to me, you're implying that stuff is there and a user can run once they build.

File: doc/UsersGuide/source/BuildingAndRunning.rst
Comment (gspetro-NOAA) [reply doc/UsersGuide/source/BuildingAndRunning.rst:108]: Ok, added some S4 info here and in Ch 3! :)

File: doc/UsersGuide/source/CodeOverview.rst
Comment (gspetro-NOAA) [reply doc/UsersGuide/source/CodeOverview.rst:23]: @DeniseWorthen Understood re: the data implications. I've added a mention that data is required. 
Generally speaking, the "Levels of Support" apply specifically to which platforms and compilers we work with (I've clarified this in the section title). We can't _guarantee_ that anything will work when using "unsupported" platforms/compilers. 
As to standalone mode, what I meant by "standalone" is “separate from an application workflow,” whereas you were (I think) interpreting "standalone" as "uncoupled." Makes sense, but not what I meant. ;) Gotta love the English language... NLP bots have their work cut out for them. :D

**Comment by DeniseWorthen** [comment]
@gspetro-NOAA Thanks the the changes, I agree. And yes, for me, "standalone" is not coupled. I think using the term "standalone" as denoting a non-workflow setup will be very confusing for most developers.

**Review by DeniseWorthen — APPROVED** [review]

**Comment by jkbk2004** [comment]
@gspetro-NOAA can you sync up branch? so that we can start getting final reviews and approvals.

**Review by zach1221 — APPROVED** [review]

**Pull request merged**
